wgsl/index.bs

```diff
@@ -90,7 +90,89 @@ that run on the GPU.
+    (For a more rigorous definition, see "[=functions in a shader stage=]".)
+* The set of variables and constants [=statically accessed=] by all those functions.
+* The set of types used to define or analyze all those functions, variables, and constants.
+
+When executing a shader stage, the implementation:
+* Binds [=resources=] to variables in the shader's [=resource interface of a shader|resource interface=],
+    making the contents of those resources available to the shader during execution.
+* Populates the formal parameters of the entry point, if they exist, with the stage's pipeline inputs.
+* Connects the entry point return value, if one exists, to the stage's pipeline outputs.
+* Then it invokes the entry point.
+
+A [SHORTNAME] program is organized into:
+* Functions, which specify execution behaviour.
+* Statements, which are declarations or units of executable behaviour.
+* Literals, which are text representations for pure mathematical values.
+* Constants, each providing a name for a value computed at a specific time.
+* Variables, each providing a name for storage for holding a value.
+* Expressions, each of which combines a set of values to produce a result value.
+* Types, each of which describes:
+    * A set of values.
+    * Constraints on supported expressions.
+    * The semantics of those expressions.
+
+[SHORTNAME] is an imperative language: behaviour is specified as a sequence of statements to execute.
+Statements:
+* Declare constants or variables
+* Modify the contents of variables
+* Modify execution order using structured programming constructs:
+    * Selective execution: if/else/elseif, switch
+    * Repetition: loop, for
+    * Escaping a nested execution construct: break, continue
+    * Refactoring: function call and return
+    * Discard (fragment shaders only): terminating the invocation and throwing away the output
+* Evaluate expressions to compute values as part of the above behaviours.
+
+[SHORTNAME] is statically typed: each value computed by a particular expression is in a specific type,
+determined only by examining the program source.
+
+[SHORTNAME] has types to describe booleans, numbers, vectors, matrices, and aggregations
+of these in the form of arrays and structures.
+Additional types describe memory.
+
+[SHORTNAME] has texture and sampler types.
+Together with their associated built-in functions, these support functionality
+commonly used for graphics rendering, and commonly provided by GPUs.
+
+The work of a shader stage is partitioned into one or more <dfn noexport>invocations</dfn>,
+each of which executes the entry point, but under slightly different conditions.
+All invocations in a shader stage share access to certain variables:
+* The resources in the shader interface.
+* Variables in the [=storage classes/workgroup=] [=storage class=], if in a [=compute shader stage|compute shader=].
```

Comment:
This is a bit misleading.  All invocations in a compute shader can access the variable, but it a different instance per workgroup.

Reply:
I'll qualify them separately. I'll pull the "All" into the first bullet point.